src/internal/rpcproviders.ts

```diff
@@ -1,84 +1,85 @@
-import {JsonRpcProvider} from "@ethersproject/providers";
+import _ from "lodash";
 
-import {ChainId, supportedChainIds} from "../common/chainid";
+import type {Provider} from "@ethersproject/providers";
 
-interface RpcProviderMap {[c: number]: JsonRpcProvider}
-
-const
-    ETH_RPC_URI_ENV:       string = "ETH_RPC_URI",
-    OPTIMISM_RPC_URI_ENV:  string = "OPTIMISM_RPC_URI",
-    CRONOS_RPC_URI_ENV:    string = "CRONOS_RPC_URI",
-    BSC_RPC_URI_ENV:       string = "BSC_RPC_URI",
-    POLYGON_RPC_URI_ENV:   string = "POLYGON_RPC_URI",
-    FANTOM_RPC_URI_ENV:    string = "FANTOM_RPC_URI",
-    BOBA_RPC_URI_ENV:      string = "BOBA_RPC_URI",
-    METIS_RPC_URI_ENV:     string = "METIS_RPC_URI",
-    MOONBEAM_RPC_URI_ENV:  string = "MOONBEAM_RPC_URI",
-    MOONRIVER_RPC_URI_ENV: string = "MOONRIVER_RPC_URI",
-    ARBITRUM_RPC_URI_ENV:  string = "ARBITRUM_RPC_URI",
-    AVALANCHE_RPC_URI_ENV: string = "AVALANCHE_RPC_URI",
-    AURORA_RPC_URI_ENV:    string = "AURORA_RPC_URI",
-    HARMONY_RPC_URI_ENV:   string = "HARMONY_RPC_URI";
+import {ChainId, supportedChainIds} from "@chainid";
+import type {StringMap} from "@common/types";
 
+import {Web3RpcConnector, JsonRpcConnector} from "./rpcconnector";
 
-const ENV_KEY_MAP: {[c: number]: string} = {
-    [ChainId.ETH]:       ETH_RPC_URI_ENV,
-    [ChainId.OPTIMISM]:  OPTIMISM_RPC_URI_ENV,
-    [ChainId.CRONOS]:    CRONOS_RPC_URI_ENV,
-    [ChainId.BSC]:       BSC_RPC_URI_ENV,
-    [ChainId.POLYGON]:   POLYGON_RPC_URI_ENV,
-    [ChainId.FANTOM]:    FANTOM_RPC_URI_ENV,
-    [ChainId.BOBA]:      BOBA_RPC_URI_ENV,
-    [ChainId.METIS]:     METIS_RPC_URI_ENV,
-    [ChainId.MOONBEAM]:  MOONBEAM_RPC_URI_ENV,
-    [ChainId.MOONRIVER]: MOONRIVER_RPC_URI_ENV,
-    [ChainId.ARBITRUM]:  ARBITRUM_RPC_URI_ENV,
-    [ChainId.AVALANCHE]: AVALANCHE_RPC_URI_ENV,
-    [ChainId.AURORA]:    AURORA_RPC_URI_ENV,
-    [ChainId.HARMONY]:   HARMONY_RPC_URI_ENV,
+const ENV_KEY_MAP: StringMap = {
+    [ChainId.ETH]:       "ETH_RPC_URI",
+    [ChainId.OPTIMISM]:  "OPTIMISM_RPC_URI",
+    [ChainId.CRONOS]:    "CRONOS_RPC_URI",
+    [ChainId.BSC]:       "BSC_RPC_URI",
+    [ChainId.POLYGON]:   "POLYGON_RPC_URI",
+    [ChainId.FANTOM]:    "FANTOM_RPC_URI",
+    [ChainId.BOBA]:      "BOBA_RPC_URI",
+    [ChainId.METIS]:     "METIS_RPC_URI",
+    [ChainId.MOONBEAM]:  "MOONBEAM_RPC_URI",
+    [ChainId.MOONRIVER]: "MOONRIVER_RPC_URI",
+    [ChainId.ARBITRUM]:  "ARBITRUM_RPC_URI",
+    [ChainId.AVALANCHE]: "AVALANCHE_RPC_URI",
+    [ChainId.AURORA]:    "AURORA_RPC_URI",
+    [ChainId.HARMONY]:   "HARMONY_RPC_URI",
 }
 
-const CHAIN_RPC_URIS: {[c: number]: string} = {
+const CHAIN_RPC_URIS: StringMap = {
```

Comment:
Do we want to allow these to be overriden in some way? The .env was cumbersome, but several networks block cors unless you run your own node. Something to consider

Reply:
If possible, I'd like it to be _slightly_ agnostic to frontend or backend; see the use of `isomorphic-fetch` in [minirpc.ts](https://github.com/synapsecns/sdk/blob/dev/src/internal/minirpc.ts#L1). 

I initially had an implementation for dynamic control of the RPC uris, but ultimately removed it -- putting it back (and rewriting it in a much better way) wouldn't be hard, though.

Reply:
Regarding CORS, I've tested this code against the frontend example (which would be making requests from localhost) and RPC functions work fine on that front.

Reply:
I'm more thinking about the cases where we've had to rapidly switch rpcs (Harmony or more recently boba)

Reply:
I actually came up with a decent implementation for setting RPC provider endpoints individually; I can also write up a function which takes a map of `{ chain id => endpoint }` and sets the endpoint for any chain in the map while leaving unpassed chains untouched

Reply:
Unfortunately, given the nature of a static NPM package, we can't necessarily update RPC endpoints all the time; we can make it very, very easy for users to set the ones being used, and provide a doc page listing a bunch of commonly used and generally working endpoints

Reply:
The latter is a better idea, ideally in production you would use your own RPC providers.

Reply:
@BlazeWasHere @trajan0x check out [51a4f87](https://github.com/synapsecns/sdk/pull/63/commits/51a4f87f019f05017975c5e5076dae561bffaa88)